File: internal/source/testdata/sample.php
<?php
/**
 * An HTTP server with configurable options.
 *
 * Supports graceful shutdown and middleware chaining.
 *
 * @since 1.0
 * @deprecated Use NewServer instead.
 */
class Server {
    /**
     * The address to listen on.
     *
     * @var string
     */
    public $addr;

    /**
     * The HTTP handler.
     *
     * @var callable|null
     */
    private $handler;

    /**
     * Creates a new Server instance.
     *
     * @param string $addr The address to listen on.
     * @param callable|null $handler The HTTP handler.
     */
    public function __construct($addr, $handler = null) {
        $this->addr = $addr;
        $this->handler = $handler;
    }

    /**
     * Starts the server on the configured address.
     *
     * @return void
     * @throws \RuntimeException If the server cannot bind.
     */
    public function listenAndServe() {
    }
}

/**
 * Creates a new Server instance with the given address.
 *
 * @param string $addr The address to listen on.
 * @return Server A new server instance.
 */
function new_server($addr) {
    return new Server($addr);
}

File: internal/source/testdata/sample.rs
/// An HTTP server with configurable options.
///
/// Supports graceful shutdown and middleware chaining.
///
/// # Examples
///
/// ```
/// let server = Server::new("127.0.0.1:8080");
/// server.listen_and_serve();
/// ```
pub struct Server {
    /// The address to listen on.
    pub addr: String,
    /// The HTTP handler.
    handler: Option<Box<dyn Fn()>>,
}

impl Server {
    /// Creates a new Server instance with the given address.
    ///
    /// # Arguments
    ///
    /// * `addr` - The address to listen on.
    pub fn new(addr: &str) -> Self {
        Server {
            addr: addr.to_string(),
            handler: None,
        }
    }

    /// Starts the server on the configured address.
    pub fn listen_and_serve(&self) -> Result<(), std::io::Error> {
        Ok(())
    }
}
